pkg/config/config.go

```diff
@@ -84,6 +85,17 @@ var defaultPathGenerator = func() (string, error) {
 // getConfigPath is the current path generator, can be replaced in tests
 var getConfigPath = defaultPathGenerator
 
+// getConfigPathWithOverride gets the config path, checking for viper override first
+func getConfigPathWithOverride() (string, error) {
+	// Check if config path is set via viper (from CLI flag)
+	if configPath := viper.GetString("config"); configPath != "" {
```

Comment:
Keep in mind that this code may not be called through the CLI interface. Should this be an env var instead?